File: 백준/Bronze/10818. 최소， 최대/최소， 최대.py
'''
1. 아이디어
- N개의 정수가 있을 때, 그 중 최대/최소를 찾는다 > min/max 함수 활용 가능
2. 시간복잡도
- min/max(): O(N)
3. 자료구조
- int N
- int[] num_list
'''

import sys
input = sys.stdin.readline

N = int(input())
num_list = list(map(int,input().split()))
print(min(num_list), max(num_list))

File: 백준/Bronze/10818. 최소， 최대/최소， 최대.java
import java.util.Scanner;
import java.util.Collections;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Main{
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        
        int N = scanner.nextInt();
        scanner.nextLine();
        String[] numStringList = scanner.nextLine().split(" ");
        List<Integer> numList = Arrays.stream(numStringList).map(Integer::parseInt).collect(Collectors.toList());
        
        int max, min;
        max = Collections.max(numList);
        min = Collections.min(numList);
        
        System.out.print(min+" "+max);
    
    }
}